incubator/hnc/pkg/reconcilers/setup.go

```diff
@@ -41,11 +41,22 @@ func Create(mgr ctrl.Manager, f *forest.Forest, maxReconciles int) error {
 		Log:                    ctrl.Log.WithName("reconcilers").WithName("HNCConfiguration"),
 		Manager:                mgr,
 		Forest:                 f,
+		Igniter:                make(chan event.GenericEvent),
 		HierarchyConfigUpdates: hcChan,
 	}
 	if err := cr.SetupWithManager(mgr); err != nil {
 		return fmt.Errorf("cannot create Config reconciler: %s", err.Error())
 	}
+	// Create a default singleton if there is no singleton in the cluster.
```

Comment:
Hmm, why not move all this into `cr.SetupWithManager`? This seems like we're leaking a bunch of internal logic about the HNCConfig reconcilers into setup.go.

Reply:
I might misunderstand your previous comment... In my previous version, I enqueued the object in SetupWithManager before the return statement. You commented "I have no reason to be scared of this, but this makes me a little scared. I feel like it might be safer to enqueue something after the reconciler's been started?" (see my comment reply above). That's why I moved it out of SetupWithManager. Did I miss anything here?

Reply:
I might misunderstand your previous comment... In my previous version, I enqueued the object in SetupWithManager before the return statement. You commented "I have no reason to be scared of this, but this makes me a little scared. I feel like it might be safer to enqueue something after the reconciler's been started?" (see my comment reply above). That's why I moved it out of SetupWithManager. Did I miss anything here?

Reply:
The return statement is what does all the real work. I just meant that instead of saying:

```
enqueue()
return ctrl.NewControllerManagedBy(mgr).A().B().Complete(r)
```

It should be something like:
```
err := ctrl.NewControllerManagedBy(mgr).A().B().Complete(r)
if err != nil {
  return err
}
enqueue()
return nil
```

A little more wordy but it keeps everything in one place.

Reply:
Thanks so much for the explanation! Feel free to take a look at the updated code.